Adding docs to interface methods [erc20/interface]

abrarabedin001 opened the pull request:
Fixes #1211


#### PR Checklist

Added documentation for erc20 interface methods.

- [x] Documentation

abrarabedin001 commented:
@ericnordelo made a new PR. I will be extra careful from now on. Thank you for guiding me.

abrarabedin001 commented:
@ericDeCourcy Hi, was there anything I needed to change here?

ericnordelo commented on packages/token/src/erc20/interface.cairo:
```diff
@@ -3,73 +3,164 @@
     fn decimals(self: @TState) -> u8;
 }
 
+/// Interface for the camelCase version of ERC20 functions.
```
```suggestion
/// Adds camelCase support for `IERC20`.
```

For consistency with the other PR

ericnordelo replied:
We have to remove individual function comments.

ericnordelo commented on packages/token/src/erc20/interface.cairo:
```diff
@@ -3,73 +3,164 @@
     fn approve(ref self: TState, spender: ContractAddress, amount: u256) -> bool;
 }
 
+/// Interface for the camelCase version of ERC20 functions without the approve function.
```
Same applies here, and we also have to remove individual function comments. Let's always keep consistency in mind.

ericnordelo commented on packages/token/src/erc20/interface.cairo:
```diff
@@ -3,73 +3,164 @@
 
 use starknet::ContractAddress;
 
+/// Interface of the ERC20 standard as defined in the EIP.
```
For consistency, let's remove the comment in the interface, and keep only the ones per function.

ericnordelo replied:
Except in camelCase where it should be added just to the interface as mentioned in another comment.

ericnordelo commented on packages/token/src/erc20/interface.cairo:
```diff
@@ -3,73 +3,164 @@
 #[starknet::interface]
 pub trait IERC20Mixin<TState> {
     // IERC20
+    /// Returns the total supply of tokens.
```
No need to comment each function for mixins, we should follow the same pattern we follow for ABIs, grouping the functions.

ericnordelo commented on packages/token/src/erc20/interface.cairo:
```diff
@@ -78,35 +169,77 @@ pub trait IERC20Permit<TState> {
 #[starknet::interface]
 pub trait ERC20ABI<TState> {
     // IERC20
+    /// Returns the total supply of tokens.
```
Similar to mixins, we don't need individual function comments for ABIs, just the grouping

ericnordelo reviewed (changes requested):
Looking good. Left a few comments focused on keeping consistency with the previous PR. We also need to be consistent in future ones.

abrarabedin001 commented:
@ericnordelo I have done the following:

- Fixed the comments regarding camelCase
- Removed comments for Minxins and ABI, just kept the grouping
- Also removed the comments right at the top of the interfaces

ericnordelo reviewed (approved):
LGTM! Thanks again for taking the time and for the patience.
